validator/length.go

```diff
@@ -0,0 +1,97 @@
+package validator
+
+import (
+	"fmt"
+
+	"github.com/spf13/cobra"
+)
+
+type LengthRule struct {
+	Use     Limit
```

Comment:
While this is clever I think it restricts us how rules are build. 
My take is that rule:

- Has name
- Has Function that does something - Doesn't have any fields 
- Has specific arguments that configure how rule works.

In this case name is defined as Struct but I think we also need name as constant.

this way we can build `LengthRule` with dictionary of key values, where key could be cobra structure key like `Use` and values might be functions, min, max etc.
```
LengthRule(Arguments{
"Use": Limit{3,10},
"Short": Limit{3,10},
})
```

Reply:
```go
type LengthRuleMap map[string]Limit // instead of struct
```
```go	
// user
var r1 validator.Rule = &validator.LengthRuleMap{
	"Use":     validator.Limit{Min: 1, Max: 5},
	"Short":   validator.Limit{Min: 4, Max: 5},
	"Long":    validator.Limit{Min: 5, Max: 20},
	"Example": validator.Limit{Min: 5, Max: 30},
}
```
yes this helps in traversing and don't break DRY
here we will need to validate the `key` in dict, to map with cobra command struct